### ISSUE TITLE
e621 scraper?

### ISSUE THREAD

**Issue by faucetlol** [Earlopain/reverser#85]
It seems a little bit of a stupid idea considering reverser is for uploading to e621, but reverser's iqdb seems to be a _lot_ better at matching image variants than e621 itself. I considered mentioning this in the past but felt like it was too silly, but was just wondering how you'd feel about it.

It would be nice to be able to scrape e621 and then have a local version of all the files to reverse image search. Having more success with finding image alts or potential duplicates that e6 iqdb doesn't give results for, provided that I have the artist's posts scraped, would especially be helpful. On the other hand, it seems incredibly pointless that every image scraped from e621 would then result in iqdb queries for an image that _we already know_ is on e621.

**Comment by Earlopain** [comment]
Fetching from e6 seems quite pointless imo. Initially I made the iqdb fork with the intent of getting it merged into e6 itself https://github.com/e621ng/e621ng/pull/368 but until now that went nowhere.

Kira has some (rightful) reservations about the new code from Danbooru but I'm thinking if it's good enough for them it should be good enough for us as well. There certainly are benefits to using the new version, like having a db that isn't prone to vanishing into thin air, or that weird sidekiq setup going on at the moment

**Comment by Earlopain** [comment]
I'm working on having the old and new version run in parallel. I tranfsered the repo to the e621ng org, still working on properly integrating it so that both versions function. Eventually there will be a checkbox to use the new version on the iqdb page. That should give me a good idea on how the results compare. Importing all the current posts will take a good chunk of time so I'll have to see how that goes.

Maybe I can even put the database into the db_export and use it locally here so I'm not constrained by the rate limit. The results would be outdated by about a day, but the sheer speed increase seems like it would be worth it.

**Comment by Earlopain** [comment]
Got it running on e9. https://e926.net/iqdb_queries?post_id=3997958&v2=1 this is currently the only post in the database. Tomorrow I'll also pull the branch on e6 and then import all the posts. After that both versions should function identically.

**Comment by faucetlol2** [comment]
Oh wow, this wasn't what I was expecting when making this issue! Very excited to try that out.

**Comment by Earlopain** [comment]
Not sure why I kept this open, should be good to close now
